private/pkg/storage/bucket.go

```diff
@@ -52,6 +52,38 @@ type ReadBucket interface {
 	Walk(ctx context.Context, prefix string, f func(ObjectInfo) error) error
 }
 
+// PutOptions are the possible options that can be passed to a Put operation.
+type PutOptions struct {
+	IfNotExists      bool
+	CustomChunkSize  bool
+	ChunkSizeInBytes int
+}
+
+// PutOption are options passed when putting an object in a bucket.
+type PutOption func(*PutOptions)
+
+// PutIfNotExists only puts the object in the bucket if there is no other object
+// already in the same bucket path. Some implementations of
+// `storage.WriteBucket.Put` that are `ReadWriteBucket` support this option.
+// Check the implementation of the bucket you're using if you're passing this
+// option.
+func PutIfNotExists() PutOption {
```

Comment:
All other options functions follow the pattern `FunctionnameWithOptionname`, ie in this case it would be `PutWithIfNotExists`. I would likely change the name of the option to  `IfNotExist`, to match `os.IsNotExist`.